test_runner/batch_others/test_pageserver_api.py

```diff
@@ -73,18 +75,39 @@ def test_pageserver_http_get_wal_receiver_success(zenith_simple_env: ZenithEnv):
     tenant_id, timeline_id = env.zenith_cli.create_tenant()
     pg = env.postgres.create_start(DEFAULT_BRANCH_NAME, tenant_id=tenant_id)
 
-    res = client.wal_receiver_get(tenant_id, timeline_id)
-    assert list(res.keys()) == [
-        "thread_id",
-        "wal_producer_connstr",
-        "last_received_msg_lsn",
-        "last_received_msg_ts",
-    ]
-
-    # make a DB modification then expect getting a new WAL receiver's data
+    def try_get_updated_msg_lsn(prev_msg_lsn: Optional[str]) -> str:
+        """Try to get the updated message's LSN from the wal receiver.
+
+        It raises an exception if the latest message's LSN is not found or
+        it hasn't been updated compared to the previous message's LSN.
+        """
+        res = client.wal_receiver_get(tenant_id, timeline_id)
+        assert list(res.keys()) == [
+            "thread_id",
+            "wal_producer_connstr",
+            "last_received_msg_lsn",
+            "last_received_msg_ts",
+        ]
+
+        if res["last_received_msg_lsn"] is None:
+            raise Exception("the last received message's LSN is empty")
+        elif prev_msg_lsn is not None and prev_msg_lsn >= res["last_received_msg_lsn"]:
```

Comment:
I'd fix the bug Heikki mentioned in the last paragraph at https://github.com/neondatabase/neon/issues/1768#issuecomment-1133752122 as well while we're here: LSNs are actually numbers with no trailing zeros and we should compare them as such.

Actually, fixing the underlying issue by altering our psycopg2 configuration would be a very nice thing to do.

Reply:
> LSNs are actually numbers with no trailing zeros and we should compare them as such.
> Actually, fixing the underlying issue by altering our psycopg2 configuration would be a very nice thing to do.

Yeah, agree. I'll try to implement the change in this PR. Not so sure about what the scope of the change will be, I guess it won't be too large.

Reply:
I think it'd be fine in a separate PR as well. This PR is concentrated in a single test, bigger PR would touch lots of LSN-related code and then some in this particular test, which may get lost in commit history.